keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,832 @@
+                                                       |---------------------------------------|
+```
 
-    string principal = 3;
-    
-    // This field is REQUIRED
-    string access_policy = 4;
-}
+#### ProvisionerCreateBucket
+
+This gRPC call creates a bucket in the OSP, and returns information about the new bucket. This api must be idempotent. The input to this call is the name of the bucket and an opaque parameters field.
+
+The returned `bucketID` should be a unique identifier for the bucket in the OSP. This value could be the name of the bucket too. This value will be used by COSI to make all subsequent calls related to this bucket.
 
-message ProvisionerGrantBucketAccessResponse {
-    // This is the account that is being provided access. This will
-    // be required later to revoke access. 
-    string principal = 1;
-    
-    string credentials_file_contents = 2;
-    
-    string credentials_file_path = 3;
-} 
 ```
+    ProvisionerCreateBucket
+    |------------------------------------------|       |-----------------------------------------------|
+    | grpc ProvisionerCreateBucketRequest{     | ===>  | ProvisionerCreateBucketResponse{              |
+    |     "name": "br-$uuid",                  |       |   "bucketID": "br-$uuid",                     |
+    |     "parameters": {                      |       |   "bucketInfo": {                             |
+    |        "key": "value"                    |       |      "s3": {                                  |
+    |     }                                    |       |        "bucketName": "br-$uuid",              |
+    | }                                        |       |        "region": "us-west1",                  |
+    | -----------------------------------------|       |        "endpoint": "s3.amazonaws.com"         |
+                                                       |      }                                        |
+                                                       |   }                                           |
+                                                       | }                                             |
+                                                       |-----------------------------------------------|
+```
+
+#### ProvisionerGrantBucketAccess
 
-## ProvisionerRevokeBucketAccess
+This gRPC call creates a set of access credentials for a bucket. This api must be idempotent. The input to this call is the id of the bucket, a set of opaque parameters and name of the account. This `accountName` field is used to ensure that multiple requests for the same BucketClaim do not result in multiple credentials.
 
-This call revokes all access to a particular bucket from a principal.  
+The returned `accountID` should be a unique identifier for the account in the OSP. This value could be the name of the account too. This value will be used by COSI to make all subsequent calls related to this account.
```

Comment:
I imagine that the provisioned account ID has more restricted permissions than an admin. Shouldn't COSI continue to use its admin-level permissions?

Reply:
The account is created for pods to access buckets (I/O path). COSI continues with its admin privileges for managing buckets (control path), not for accessing them (I/O path).

Reply:
The sentence "This value will be used by COSI to make all subsequent calls related to this account." makes it sound like COSI is going to be using that account for its own calls. It sounds like that is not the case and COSI will use it to manage any future changes to bucket access.

Reply:
Good point. I'll update it

Reply:
Don't see an update?

Reply:
Apologies, I made a mistake while pushing the latest changes to this PR. The current version has all the updates.